### PR TITLE
Create DappNode package and document process

### PR DESCRIPTION
**Pull request by cburgdorf**
### What was wrong?

DappNode is a popular "operating system" to run various kinds of decentralized services. Exposing Trinity as a DappNode package has several advantages:

- Making it easier for people to try out Trinity
- Increasing awareness of the Trinity client 
- Giving Trinity developers a simple way to perform long running experiments at home
- ...

### How was it fixed?

- Added a new `make` entry to cut DappNode releases
- Added the needed DappNode package scaffolding
- Added docs on how to cut DappNode releases
- Added docs on how to install and use Trinity as a DappNode package

**Preview of the guide in action** :point_right:  https://external-builds.readthedocs.io/html/trinity-client/1082/guides/running_trinity_on_dappnode.html

[//]: # (Stay ahead of things, add list items here!)
- [x] Clean up commit history

[//]: # (For important changes that should go into the release notes please add a newsfragment file as explained here: https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

[//]: # (See: https://trinity-client.readthedocs.io/en/latest/contributing.html#pull-requests)
- [x] Add entry to the [release notes](https://github.com/ethereum/trinity/blob/master/newsfragments/README.md)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -86,6 +86,11 @@ release: clean
 create-docker-image: clean
 	docker build -t ethereum/trinity:latest -t ethereum/trinity:$(version) .
 
+create-dappnode-image: clean
+	sed -i -e 's/ARG GITREF=\w*/ARG GITREF=$(trinity_version)/g' ./dappnode/build/Dockerfile
```
Comment (pipermerriam) [inline Makefile]: my experience has been that `sed` behaves differently on OSX/linux sometimes.  Have you verified this works on both?

File: dappnode/trinity.dnp.dappnode.eth.env [hunk 2]
```diff
@@ -0,0 +1 @@
+EXTRA_OPTS--trinity-root-dir /trinity --sync-mode beam
```
Comment (pipermerriam) [inline dappnode/trinity.dnp.dappnode.eth.env]: I think the `--sync-mode beam` isn't needed any longer.

Comment (cburgdorf) [reply dappnode/trinity.dnp.dappnode.eth.env]: Good catch, will drop it.

File: Makefile (hunk 1 above)
Comment (cburgdorf) [reply Makefile]: I don't have a mac around but I'll suppose we'll find it out quickly if anyone with a Mac (do we even have Mac users within our Trinity crew :thinking:) wants to get into cutting DappNode releases :)

**Pull request merged**